Add max length configuration for connection info in status bar

bathetrade opened the pull request:
Allows configuring the display width of the connection info in the status bar.

This makes it possible to extend the width and see the entire server name, database name, and user name without needing to hover over the item with the mouse. This in turn reduces the likelihood that a query is issued to the wrong database and may be more familiar for SSMS users where such information is always displayed in the title bar.

bathetrade commented:
@microsoft-github-policy-service agree

kburtram commented:
@Benjin what do you think of this one?  It looks fine to me.  Though maybe the default should be -1, or 100.  Or maybe 50 is enough for most people?

bathetrade commented:
@kburtram The default width before this PR was 50 (Constants.maxDisplayedStatusTextLength), so I just left it at that. The servers, databases, and usernames I regularly work with however aren't particularly long and are truncated enough that a good amount of the database name is hidden, so I imagine increasing the default would be a welcome change.

Benjin commented on src/models/connectionInfo.ts:
```diff
@@ -149,9 +150,19 @@ export function getConnectionDisplayString(creds: IConnectionInfo): string {
+        .getConfiguration(Constants.extensionConfigSectionName)
+        .get(Constants.configStatusBarConnectionInfoMaxLength);
+
+    statusBarConnectionInfoMaxLength ||= Constants.maxDisplayedStatusTextLength;
```
If the config value has a default, this'll never be undefined, so you can just drop this line and delete `maxDisplayedStatusTextLength` from `Constants`.

bathetrade replied:
Done. I bumped the default up to 100 as well.

Benjin commented:
@kburtram agreed; this is a nice addition!  I left one minor comment, then I'd be happy to take it.

Benjin commented:
Sorry for taking a while to circle back to this; @bathetrade, can you resolve the conflicts?  Then I'll be ready to merge it in :)

bathetrade commented:
@Benjin I resolved the merge conflicts and made a couple more changes:

1. Removed nesting of `connectionInfo` test suites since the "getConnectionDisplayString" suite wasn't running or showing up in the test explorer for me
2. Moved VS Code configuration loading out of `getConnectionDisplayString` to make it easier to test and more flexible for callers
3. Updated the README to reflect the new setting
4. Changed the default length to -1 instead of 100 in light of #19485 and #18865

While testing, I ran into the following issues (unrelated to this PR)
1. Debug with "Launch Extension (With Other Extensions Disabled)"
2. Open a new file and change the type to SQL
3. Press `ctrl+shift+c` to open the connection picker and select a profile
4. ![image](https://github.com/user-attachments/assets/6171d79d-3980-49aa-b19f-1228cb7635ef)
5. Open the extension settings from the extension's cogwheel in the sidebar
6. Change a setting (I used the new status bar setting, but others seem to have the same behavior)
7. Click somewhere outside of the setting area to commit the change
8. ![image](https://github.com/user-attachments/assets/b5472d2d-b479-4f2c-ae2b-7ae82d66d429)

I tested off `main` and was getting the same behavior but wanted to mention it anyway in case it's not yet being tracked.

Another issue is that the display string is now using icon tokens such as "$(database)". If you specify a max length that lands in the middle of such a token, the user will see something like "myServer: $(data ..." in the status bar. I didn't have time to fix it but would be happy to at some point if deemed worthwhile.

swaschitz commented:
As an update since this feature/change doesn't appear done, with the release of 1.33.0, the default behavior has actually gotten worse.  It now truncates earlier.  

<img width="540" alt="Image" src="https://github.com/user-attachments/assets/e4d56e6d-bc84-4ef7-8890-391e3986af8b" />
